store/store.go

```diff
@@ -302,6 +302,46 @@ func (s *Store[H]) HasAt(_ context.Context, height uint64) bool {
 	return height != uint64(0) && s.Height() >= height
 }
 
+// DeleteRange implements [header.Store] interface.
+func (s *Store[H]) DeleteRange(ctx context.Context, from, to uint64) error {
+	if from >= to {
+		return fmt.Errorf("header/store: invalid range(%d,%d)", from, to)
+	}
+
+	last, err := s.GetByHeight(ctx, to-1)
```

Comment:
Should we extract getting headers hashes as a separate func? Looks a bit too long now.

Reply:
Nah, much simpler now (and correct).